board/batocera/rockchip/rk3326/linux_patches/linux-004-rg351-controller-rumble.patch

```diff
@@ -0,0 +1,290 @@
+diff --git a/drivers/input/joystick/Kconfig b/drivers/input/joystick/Kconfig
+index 5e38899..3c5eb90 100644
+--- a/drivers/input/joystick/Kconfig
++++ b/drivers/input/joystick/Kconfig
+@@ -15,6 +15,9 @@ menuconfig INPUT_JOYSTICK
+ 
+ if INPUT_JOYSTICK
+ 
++config JOYSTICK_ADC_GPIO
++	tristate "ADC Analog + GPIO keys. Polled."
++
+ config JOYSTICK_ANALOG
+ 	tristate "Classic PC analog joysticks and gamepads"
+ 	select GAMEPORT
+diff --git a/drivers/input/joystick/Makefile b/drivers/input/joystick/Makefile
+index 31d720c..1d55c42 100644
+--- a/drivers/input/joystick/Makefile
++++ b/drivers/input/joystick/Makefile
+@@ -40,3 +40,4 @@
+ obj-$(CONFIG_JOYSTICK_WALKERA0701)	+= walkera0701.o
+ obj-$(CONFIG_JOYSTICK_XPAD)		+= xpad.o
+ obj-$(CONFIG_JOYSTICK_ZHENHUA)		+= zhenhua.o
++obj-$(CONFIG_JOYSTICK_ADC_GPIO)         += adc-gpio.o
+diff --git a/linux-go-next/drivers/input/joystick/adc-gpio.c b/drivers/input/joystick/adc-gpio.c
+new file mode 100644
+index 0000000..ee8dc1f
+--- /dev/null
++++ b/drivers/input/joystick/adc-gpio.c
+@@ -0,0 +1,260 @@
++// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
++/*
++ * Copyright (C) 2021, Anonymous
```

Comment:
It's `Maya Matuszczyk <maccraft123mc@gmail.com>`, not Anonymous.
https://github.com/Maccraft123/linux/blob/mac/go-next/drivers/input/joystick/adc-gpio.c